client/app/pages/ItemsList.jsx

```diff
@@ -0,0 +1,320 @@
+
+    this.state.paginator.rowsFetcher = (
+      requestedPage,
+      itemsPerPage,
+      orderByField,
+      orderByReverse,
+      paginator,
+      requested = false,
+    ) => {
+      // in search mode ignore the ordering and use the ranking order
+      // provided by the server-side FTS backend instead, unless it was
+      // requested by the user by actively ordering in search mode
+      if (this.isInSearchMode && !requested) {
+        orderByField = undefined;
+      }
+      if (orderByField && orderByReverse) {
+        orderByField = this.constructor.orderSeparator + orderByField;
+      }
+
+      $location.search({
+        page: requestedPage,
+        page_size: itemsPerPage,
+        order: orderByField,
+        q: this.state.searchTerm !== '' ? this.state.searchTerm : null,
+      });
+
+      this.setState({ isLoaded: false });
+      const request = this.getRequest(requestedPage, itemsPerPage, orderByField);
+      return this.doRequest(request).then(requestCallback);
+    };
+
+    this.updatePaginator();
+  }
+
+  // eslint-disable-next-line class-methods-use-this, no-unused-vars
+  onRowClick(event, item) {
+  }
+
+  get isInSearchMode() {
+    return isString(this.state.searchTerm) && (this.state.searchTerm !== '');
+  }
+
+  getSidebarMenu() {
+    return this.constructor.sidebarMenu;
+  }
+
+  getListColumns() {
+    return this.constructor.listColumns;
+  }
+
+  getRequest(requestedPage, itemsPerPage, orderByField) {
+    const request = {
+      page: requestedPage,
+      page_size: itemsPerPage,
+      order: orderByField,
+      tags: this.state.selectedTags,
+    };
+    if (this.isInSearchMode) {
+      request.q = this.state.searchTerm;
+    }
+    return request;
+  }
+
+  toggleSorting(field) {
+    this.state.paginator.orderBy(field);
+  }
+
+  // eslint-disable-next-line no-unused-vars, class-methods-use-this
+  doRequest(request) {
+    return null;
+  }
+
+  processResponse() {
+    this.setState({ isLoaded: true });
+  }
+
+  update(state) {
+    this.setState(state, () => {
+      this.updatePaginator();
+    });
+  }
+
+  updatePaginator(updates) {
+    const paginator = this.state.paginator;
+    updates = extend({ page: paginator.page }, updates);
+    paginator.setPage(
+      updates.page,
+      updates.pageSize,
+      updates.pageOrder,
+      updates.pageOrderReverse,
+    );
+    // this.forceUpdate();
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderLoadingState() {
+    return (
+      <div className="text-center">
+        <BigMessage icon="fa-spinner fa-2x fa-pulse" message="Loading..." />
+      </div>
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  renderEmptyState() {
+    return (
+      <div className="text-center">
+        <BigMessage icon="fa-search" message="Sorry, we couldn't find anything." />
+      </div>
+    );
+  }
+
+  renderSearchInput(placeholder = null) {
+    return (
+      <div className="m-b-10">
+        <Input
```

Comment:
If you use `<Input.Search` then a search icon is added to the side. Could be nice.

Reply:
Yep, I tried it. But this input is already small, and icon will just use an extra space.